common/ResetPasswordCore.py

```diff
@@ -0,0 +1,177 @@
+            return('FAILED')
+
+    @staticmethod
+    def __validate_reset_password_token(token: str) -> int:
+        """ Return user's ID if reset password token is correct. """
+        hash = Serializer(SECRET_KEY)
+        try:
+            data = hash.loads(token)
+        except SignatureExpired:
+            raise
+        except BadSignature:
+            raise
+
+        return(data['id'])
+
+    @staticmethod
+    def __verify_user_input_with_hashed_data(hashed_data: str,
+                                             user_input: str) -> int:
+        """ Verify if user's inputed 8 digit code is correct, if so,
+        return users id."""
+        hash = Serializer(SECRET_KEY)
+        try:
+            data = hash.loads(hashed_data)
+        except SignatureExpired:
+            raise
+        except BadSignature:
+            raise
+
+        if user_input == data['valid_code']:
+            return(data['id'])
+        else:
+            return(-1)
+
+    @staticmethod
+    def __generate_reset_password_token(id: int) -> str:
+        """ Generate the main token that reset user's password."""
+        hash = Serializer(SECRET_KEY, expires_in=300)
+        return(
+            str(
+                hash.dumps({
+                    'id': id
+                }),
+                'utf-8'
+            )
+        )
+
+    @staticmethod
+    def __generate_hash_string(id: int, random_code: str) -> str:
+        """ Create a hash that contains (user's ID and valid
+        random created 8 digit code) which expires in 300 seconds."""
+        hash = Serializer(SECRET_KEY, expires_in=300)
+        return(
+            str(
+                hash.dumps({
+                    'id': id,
+                    'valid_code': random_code
+                }),
+                'utf-8'
+            )
+        )
+
+    @staticmethod
+    def __send_email_to_user(username: str, subject: str, msg_body: Text):
+        """ Send Email to username's Email. msg_body is Jinja2
+        rendered template."""
+        host = CONFIG.get('smtp', {}).get('host')
+        port = CONFIG.get('smtp', {}).get('port')
+        sender_email = CONFIG.get('smtp', {}).get('email')
+        password = CONFIG.get('smtp', {}).get('password')
+
+        # For passing CI tests
+        if host == 'your_host_address_here':
+            return
+
+        smtp_server = smtplib.SMTP(host=host, port=port)
+        smtp_server.starttls()
+        smtp_server.login(sender_email, password)
+
+        msg = MIMEMultipart()
+        msg['From'] = sender_email
+        msg['To'] = username
+        msg['Subject'] = subject
+        body_rendered = msg_body
+        msg.attach(MIMEText(body_rendered, 'html'))
+
+        smtp_server.send_message(msg)
+        smtp_server.quit()
+
+    @staticmethod
+    def __generate_8_digit_code() -> str:
+        """ For generating random 8 digit of integers."""
+        result_str = ''.join(
+            random.choice(string.digits) for i in range(8)
```

Comment:
Did you know instead of for loop you can use:
`random.choices(string.digits, k=length)`